sound/soc/sof/intel/hda-dsp.c

```diff
@@ -398,8 +398,13 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 	if (target_state->substate == SOF_HDA_DSP_D0I3) {
 		value = SOF_HDA_VS_D0I3C_I3;
 
-		/* disable DMA trace in D0I3 */
-		flags = HDA_PM_NO_DMA_TRACE;
+		/*
+		 * Trace DMA can be kept enabled when the DSP enters
+		 * D0I3 while the system is in S0. It must be disabled
+		 * when the system is suspending to S0IX.
```

Comment:
but by default don't we want to disable the trace in D0i3/S0 cases?

This will lead to a different behavior and make it complicated to debug trace-related issues?

Reply:
@plbossart I think keeping trace enabled by default in D0I3/S0 makes more sense. If we do suspect trace-related issues, we have the option of disabling trace entirely no?

Reply:
@ranj063 I would rather require the developer to enable the trace in D0ix/S0 if they really want it. We should really try and use D0ix in the same configuration by default, regardless of the S state we are in.

Reply:
@plbossart OK, so should I make this configurable in Kconfig?

Reply:
probably better with a kernel parameter, this isn't something that distros should know about, only expert users.

Reply:
@plbossart fixed now